webapp/cas-server-webapp-config/src/main/java/org/apereo/cas/web/security/CasWebSecurityConfigurerAdapter.java

```diff
@@ -278,11 +279,24 @@ protected void configureEndpointAccessByIpAddress(final AuthorizeHttpRequestsCon
                 val remoteAddr = StringUtils.defaultIfBlank(
                     context.getRequest().getHeader(casProperties.getAudit().getEngine().getAlternateClientAddrHeaderName()),
                     context.getRequest().getRemoteAddr());
+
+                LOGGER.trace("Attempting to match [{}] against [{}] as a IP or netmask", remoteAddr, addresses);
+                val addressNets = properties.getRequiredIpAddresses()
+                        .stream()
+                        .filter(addr -> !addr.contains(".+"))
```

Comment:
This can not work. You should parsing the pattern with the assumption that it's valid regex. If you can get a pattern, then you move forward. If not, then you treat it as non-regex. Can't check for specific patterns or assumptions.

Reply:
Or the opposite: parse the value with the assumption that it's valid IP or subnet, and if that fails, treat it as regex.  

https://github.com/apereo/cas/blob/6.5.x/webapp/cas-server-webapp-config/src/main/java/org/apereo/cas/web/security/CasWebSecurityExpressionRoot.java#L27-L41

Reply:
This might be useful https://commons.apache.org/proper/commons-net/apidocs/org/apache/commons/net/util/SubnetUtils.html#SubnetUtils-java.lang.String-

Reply:
I was just filtering out a couple common regular expressions to avoid the parsing that goes on when `.filter(addr -> FunctionUtils.doWithoutThrows(ip -> new IpAddressMatcher(addr)))` runs and filters out the rest - but I realize it's hacky and I can probably combine both checks into one stream.

Reply:
I updated it so it makes one pass through the lists of addresses and if they aren't valid IPs or netmasks then it treats them as regular expressions. Once this is OK I would like to backport to 6.6.x.

Reply:
Sounds good, sure.